packages/yew/src/virtual_dom/vlist.rs

```diff
@@ -156,27 +156,51 @@ mod test {
 
 #[cfg(feature = "ssr")]
 mod feat_ssr {
+    use futures::stream::{FuturesOrdered, StreamExt};
+
     use super::*;
     use crate::html::AnyScope;
+    use crate::io::{self, BufWriter};
 
     impl VList {
-        pub(crate) async fn render_to_string(
+        pub(crate) async fn render_into_stream(
             &self,
-            w: &mut String,
+            w: &mut BufWriter,
             parent_scope: &AnyScope,
             hydratable: bool,
         ) {
-            // Concurrently render all children.
-            for fragment in futures::future::join_all(self.children.iter().map(|m| async move {
-                let mut w = String::new();
+            if self.children.len() < 2 {
```

Comment:
Nit: this is cleaner and easier to read as:
```rust
match self.children.len() {
0 => {}
1 => { self.children.first().unwrap(); ... }
_ => ...
```
It's not that the current implementation is wrong in any way. I found that the equality check, match and early return are harder to read compared to a single match

Reply:
I think it'd even easier to read with a [slice pattern](https://doc.rust-lang.org/reference/patterns.html#slice-patterns). `self.children` isn't consumed, is it?